client/src/chat/MessageBubble.jsx

```diff
@@ -0,0 +1,181 @@
+            </div>
+            {reactions && reactions.length > 0 && (
+              <ReactionList
+                reactions={reactions}
+                players={players}
+                currentPlayerPosition={currentPlayerPosition}
+                onRemove={onRemoveReaction}
+              />
+            )}
+          </div>
+
+          {canReact && isHovered && !showEmojiPicker && (
+            <button
+              type="button"
+              className="absolute -left-8 top-0 p-1 rounded-full hover:bg-gray-200 transition-opacity opacity-0 hover:opacity-100 animate-fadeIn"
+              onClick={() => setShowEmojiPicker(!showEmojiPicker)}
+              title="Add reaction"
+              aria-label="Add reaction"
+            >
+              <EmojiIcon className="h-5 w-5 text-gray-600" />
+            </button>
+          )}
+
+          {showEmojiPicker && (
+            <div className="absolute right-0 top-0">
+              <EmojiPicker
+                emojis={reactionEmojisAvailable}
+                onSelect={handleEmojiSelect}
+                position="above"
+              />
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  // Left-aligned gray bubble for other messages
+  return (
+    <div className="flex items-start my-2 px-2">
+      <div className="mr-2">{avatarImage}</div>
+      <div
+        className="relative max-w-[75%]"
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => {
+          setIsHovered(false);
+          setShowEmojiPicker(false);
+        }}
+      >
+        <div className="flex flex-col gap-1">
+          {(showNickname || showTitle) && (
+            <p className="text-xs px-2">
+              {showNickname && (
+                <span className="font-semibold text-gray-900">
+                  {sender.name}
+                </span>
+              )}{" "}
+              {showTitle && (
+                <span
+                  className={`${
+                    showNickname
+                      ? "font-normal text-gray-500"
+                      : "font-semibold text-gray-900"
+                  }`}
+                >
+                  {showNickname ? `(${sender.title})` : sender.title}
+                </span>
+              )}
+            </p>
+          )}
+          <div className="bg-gray-200 text-gray-900 rounded-2xl px-4 py-2 break-words">
+            <p className="text-sm leading-relaxed">{text}</p>
+          </div>
+          <div className="text-xs text-gray-400 px-2">
+            {createdAt && relTime(createdAt)}
+          </div>
+          {reactions && reactions.length > 0 && (
+            <ReactionList
+              reactions={reactions}
+              players={players}
+              currentPlayerPosition={currentPlayerPosition}
+              onRemove={onRemoveReaction}
+            />
+          )}
+        </div>
+
+        {canReact && isHovered && !showEmojiPicker && (
+          <button
```

Comment:
Lets have the emoji picker icon button be always there, actually, so that it's more obvious for participants. But, lets make it fairly light, and then darken when the player hovers over anything in this component.

Reply:
Also, currently the emoji picker icon shows up when I hover over the bubble, but then immediately disappears, even if the mouse is still over the bubble. Im guessing that the isHovered is getting turned off too soon?